contrib/toxfile.py

```diff
@@ -0,0 +1,106 @@
+from pathlib import Path
+
+from tox.execute.request import StdinSource
+from tox.plugin import impl
+
+
+CWD = Path().resolve()
+HERE = Path(__file__).parent.resolve()
+
+if CWD != HERE.parent:
+    raise RuntimeError("Tox must be run from the sopel repository root")
+
+
+@impl
+def tox_on_install(tox_env, arguments, section, of_type):
+    EXCLUDED_DEPS = ("sopel-help",)
+
+    # NOTE: this hook is a workaround for the cyclic dependency between sopel and sopel-help
+    if of_type == "deps":
+        cmd = ["python", "-m", "pip", "install", "--no-deps", "sopel-help"]
+        tox_env.execute(cmd, stdin=StdinSource.OFF)
+    elif of_type == "package":
+        sopel_sdist = arguments[0]
+        deps = sopel_sdist.deps
+        deps[:] = [d for d in deps if d.name not in EXCLUDED_DEPS]
+
+
+# NOTE:The following is included verbatim from upstream (v0.2.0.post2 @ ebed159)
+# to allow re-use of tox environments for each Python version
```

Comment:
I see upstream specifically recommends vendoring that plugin to avoid breakage. Maybe mention that recommendation here, so it's clearer why this code was pasted into `toxfile.py` instead of just documenting the plugin as recommended for users of the `tox` config to install?